app/controllers/bid_acceptances_controller.rb

```diff
@@ -1,9 +1,12 @@
 class BidAcceptancesController < ApplicationController
-	before_action :authenticate_broker_or_forward_freight!
+	before_action :authenticate_business_user!
 
   def create
     @bid = Bid.find(params[:bid_id])
     @bid.toggle_accept(@bid.accepted)
+    
+    @bid.accepted ? @bid.job.update(agent_id: @bid.bidder.id) : @bid.job.update(agent_id: nil)
```

Comment:
Let's put this in the `toggle_accept` method in the `Bid` class.

```
def toggle_accept!(status)
  self.accepted = !status
  self.agent_id = accepted ? bidder.id : nil
  self.save
end
```

Notice the bang operator - since we are changing values, it's common practice to add a `!` to the end of the method :)